precompiles/assets-erc20/src/mock.rs

```diff
@@ -0,0 +1,285 @@
+	construct_runtime, parameter_types,
+	traits::{AsEnsureOriginWithArg, Everything},
+	weights::Weight,
+};
+
+use frame_system::{EnsureNever, EnsureRoot};
+use pallet_evm::{EnsureAddressNever, EnsureAddressRoot};
+use precompile_utils::{
+	mock_account,
+	precompile_set::*,
+	testing::{AddressInPrefixedSet, MockAccount},
+};
+use sp_core::H256;
+use sp_runtime::{
+	traits::{BlakeTwo256, ConstU32, IdentityLookup},
+	BuildStorage,
+};
+
+pub type AccountId = MockAccount;
+pub type AssetId = u128;
+pub type Balance = u128;
+pub type Block = frame_system::mocking::MockBlockU32<Runtime>;
+
+/// The foreign asset precompile address prefix. Addresses that match against this prefix will
+/// be routed to Erc20AssetsPrecompileSet being marked as foreign
+pub const FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX: u32 = 0xffffffff;
+
+parameter_types! {
+	pub ForeignAssetPrefix: &'static [u8] = &[0xff, 0xff, 0xff, 0xff];
+}
+
+mock_account!(ForeignAssetId(AssetId), |value: ForeignAssetId| {
+	AddressInPrefixedSet(FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX, value.0).into()
+});
+
+// Implement the trait, where we convert AccountId to AssetID
+impl AccountIdAssetIdConversion<AccountId, AssetId> for Runtime {
+	/// The way to convert an account to assetId is by ensuring that the prefix is 0XFFFFFFFF
+	/// and by taking the lowest 128 bits as the assetId
+	fn account_to_asset_id(account: AccountId) -> Option<(Vec<u8>, AssetId)> {
+		if account.has_prefix_u32(FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX) {
+			return Some((
+				FOREIGN_ASSET_PRECOMPILE_ADDRESS_PREFIX
+					.to_be_bytes()
+					.to_vec(),
+				account.without_prefix(),
+			));
+		}
+		None
+	}
+
+	// Not used for now
+	fn asset_id_to_account(_prefix: &[u8], asset_id: AssetId) -> AccountId {
+		ForeignAssetId(asset_id).into()
+	}
+}
+
+parameter_types! {
+	pub const BlockHashCount: u32 = 250;
+	pub const SS58Prefix: u8 = 42;
+}
+
+impl frame_system::Config for Runtime {
+	type BaseCallFilter = Everything;
+	type DbWeight = ();
+	type RuntimeOrigin = RuntimeOrigin;
+	type Nonce = u64;
+	type Block = Block;
+	type RuntimeCall = RuntimeCall;
+	type Hash = H256;
+	type Hashing = BlakeTwo256;
+	type AccountId = AccountId;
+	type Lookup = IdentityLookup<Self::AccountId>;
+	type RuntimeEvent = RuntimeEvent;
+	type BlockHashCount = BlockHashCount;
+	type Version = ();
+	type PalletInfo = PalletInfo;
+	type AccountData = pallet_balances::AccountData<Balance>;
+	type OnNewAccount = ();
+	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
+	type BlockWeights = ();
+	type BlockLength = ();
+	type SS58Prefix = SS58Prefix;
+	type OnSetCode = ();
+	type MaxConsumers = frame_support::traits::ConstU32<16>;
+	type RuntimeTask = ();
+}
+
+parameter_types! {
+	pub const MinimumPeriod: u64 = 5;
+}
+
+impl pallet_timestamp::Config for Runtime {
+	type Moment = u64;
+	type OnTimestampSet = ();
+	type MinimumPeriod = MinimumPeriod;
+	type WeightInfo = ();
+}
+
+parameter_types! {
+	pub const ExistentialDeposit: u128 = 0;
```

Comment:
I also think you will need to change this to something that is not `0`, otherwise running the test individually for this crate will give `The existential deposit must be greater than zero!`.
```suggestion
	pub const ExistentialDeposit: u128 = 1;
```

Reply:
As we are testing in an ethereum-based context where the existential deposit is `0`, I think it makes sense to leave it configured as it is. In any case I can add the `insecure-zero-ed` flag to pallet-balances in `dev` dependencies for tests to run separately.

Reply:
As we are testing in an ethereum-based context where the existential deposit is `0`, I think it makes sense to leave it configured as it is. In any case I can add the `insecure_zero_ed` flag to `pallet-balances` for tests to run separately, what do you think?

Reply:
Sounds good, we already do it in a few places.

Reply:
Also since we only use `pallet-balances` in the mock, we could move it as dev depencency.